drivers/SmartThings/matter-lock/src/new-matter-lock/init.lua

```diff
@@ -0,0 +1,1580 @@
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+local device_lib = require "st.device"
+local capabilities = require "st.capabilities"
+local clusters = require "st.matter.clusters"
+local im = require "st.matter.interaction_model"
+local lock_utils = require "lock_utils"
+
+local version = require "version"
+if version.api < 10 then
+  clusters.DoorLock = require "DoorLock"
+end
+
+local DoorLock = clusters.DoorLock
+local INITIAL_COTA_INDEX = 1
+local ALL_INDEX = 0xFFFE
+
+local NEW_MATTER_LOCK_PRODUCTS = {
+  {0x115f, 0x2802}, -- AQARA, U200
+  {0x115f, 0x2801}, -- AQARA, U300
+  {0x10E1, 0x2002} -- VDA
+}
+
+local subscribed_attributes = {
+  [capabilities.lock.ID] = {
+    DoorLock.attributes.LockState
+  },
+  [capabilities.remoteControlStatus.ID] = {
+    DoorLock.attributes.OperatingMode
+  },
+  [capabilities.lockUsers.ID] = {
+    DoorLock.attributes.NumberOfTotalUsersSupported
+  },
+  [capabilities.lockCredentials.ID] = {
+    DoorLock.attributes.NumberOfPINUsersSupported,
+    DoorLock.attributes.MaxPINCodeLength,
+    DoorLock.attributes.MinPINCodeLength,
+    DoorLock.attributes.RequirePINforRemoteOperation
+  },
+  [capabilities.lockSchedules.ID] = {
+    DoorLock.attributes.NumberOfWeekDaySchedulesSupportedPerUser,
+    DoorLock.attributes.NumberOfYearDaySchedulesSupportedPerUser
+  }
+}
+
+local subscribed_events = {
+  [capabilities.lock.ID] = {
+    DoorLock.events.LockOperation
+  },
+  [capabilities.lockAlarm.ID] = {
+    DoorLock.events.DoorLockAlarm
+  },
+  [capabilities.lockUsers.ID] = {
+    DoorLock.events.LockUserChange
+  }
+}
+
+local function is_new_matter_lock_products(opts, driver, device)
+  if device.network_type ~= device_lib.NETWORK_TYPE_MATTER then
+    return false
+  end
+  for _, p in ipairs(NEW_MATTER_LOCK_PRODUCTS) do
+    if device.manufacturer_info.vendor_id == p[1] and
+      device.manufacturer_info.product_id == p[2] then
+        return true
+    end
+  end
+  return false
+end
+
+local function find_default_endpoint(device, cluster)
+  local res = device.MATTER_DEFAULT_ENDPOINT
+  local eps = device:get_endpoints(cluster)
+  table.sort(eps)
+  for _, v in ipairs(eps) do
+    if v ~= 0 then --0 is the matter RootNode endpoint
+      return v
+    end
+  end
+  device.log.warn(string.format("Did not find default endpoint, will use endpoint %d instead", device.MATTER_DEFAULT_ENDPOINT))
+  return res
+end
+
+local function component_to_endpoint(device, component_name)
+  return find_default_endpoint(device, clusters.DoorLock.ID)
+end
+
+local function device_init(driver, device)
```

Comment:
You'll need to make this change as I suggested in the comment here: https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1624#discussion_r1791044889

```suggestion
local function device_init(driver, device)
local function device_init(driver, device)
  device:set_component_to_endpoint_fn(component_to_endpoint)
  for cap_id, attributes in pairs(subscribed_attributes) do
    if device:supports_capability_by_id(cap_id) then
      for _, attr in ipairs(attributes) do
        device:add_subscribed_attribute(attr)
      end
    end
  end
  for cap_id, events in pairs(subscribed_events) do
    if device:supports_capability_by_id(cap_id) then
      for _, e in ipairs(events) do
        device:add_subscribed_event(e)
      end
    end
  end
  device:subscribe()
 end
```